### ISSUE TITLE
[Feature Request] RGB to YUV and YUV to RGB

### ISSUE THREAD

**Issue by ananyamukh6** [kornia/kornia#297]
Here is a prototype branch.
Using out_channel=3 X inchannel=3 X 1 X 1 conv to implement yuv to rgb and rgb to yuv

https://github.com/kornia/kornia/compare/master...ananyamukh6:ananyamukh6/yuv

@edgarriba @ducha-aiki , do you think this looks useful? If so I can add tests/docs etc, polish it up and submit a PR

**Comment by edgarriba** [comment]
@ananyamukh6 awesome. However, regarding your implementation I wouldn't use conv2d, we benchmarked before and seems more efficient just dot products. Please, feel free to open a PR

**Comment by ananyamukh6** [comment]
Interesting comment about the conv vs dot. Do you mean I should do:
```
    w1 = torch.tensor([0.299, 0.587, 0.114])
    w2 = torch.tensor([-0.14713, -0.28886, 0.436])
    w3 = torch.tensor([0.615, -0.51499, 0.10001])
    y = torch.tensordot(input, w1, dims=([1], [0])).unsqueeze(1)
    u = torch.tensordot(input, w2, dims=([1], [0])).unsqueeze(1)
    v = torch.tensordot(input, w3, dims=([1], [0])).unsqueeze(1)
    return torch.cat((y,u,v), 1)
```
instead of:
```
weights = torch.tensor([[[[0.299]], [[0.587]], [[0.114]]], [[[-0.14713]], [[-0.28886]], [[0.436]]], [[[0.615]], [[-0.51499]], [[0.10001]]]])
return F.conv2d(input, weights)
```

**Comment by edgarriba** [comment]
@ananyamukh6 I suggest you to try both and benchmark. Any reason why using tensodot instead of matmul ?

just checking from wiki: https://es.wikipedia.org/wiki/YUV

I would try both forms and benchmark
1) Split rgb and do mul + sum element wise 
2) Similar to what you did but just simple dot product. For this one you can create a (*,3,3) weights tensor and do at once with matmul or (@ op)

**Comment by ananyamukh6** [comment]
I am new to pytorch so am learning on the way. Thanks for pointing out `torch.matmul`. I'll try that out as well

**Comment by ananyamukh6** [comment]
Running experiments of a `[4, 3, 243, 410]` image

# Method 1: Using conv2d: 0.03423714637756348
# weights of size 3 x 3 x 1 x 1
    weights = torch.tensor([[[[0.299]], [[0.587]], [[0.114]]], [[[-0.14713]], [[-0.28886]], [[0.436]]], [[[0.615]], [[-0.51499]], [[0.10001]]]])
    return F.conv2d(input, weights)


# Method 2: Using tensordot: 0.0197908878326416
    w1 = torch.tensor([0.299, 0.587, 0.114])
    w2 = torch.tensor([-0.14713, -0.28886, 0.436])
    w3 = torch.tensor([0.615, -0.51499, 0.10001])
    y = torch.tensordot(input, w1, dims=([1], [0]))
    u = torch.tensordot(input, w2, dims=([1], [0]))
    v = torch.tensordot(input, w3, dims=([1], [0]))
    return torch.cat((y,u,v), 1)

# Method 3: Using matmul: 0.01232290267944336
    input = input.transpose(-1,-3) # NCHW -> NHWC, or CHW -> HWC
    weights = torch.tensor([[0.299, -0.14713, 0.615], [0.587, -0.28886, -0.51499], [0.114, 0.436, 0.10001]])
    return torch.matmul(input, weights).transpose(-1,-3) # restore to CHW or NCHW
    

# Method 4: split, mul, add, join: 0.007370948791503906
    r, g, b = torch.chunk(input, chunks=3, dim=-3)
    y = 0.299*r + 0.587*g + 0.114*b
    u = -0.14713*r + -0.28886*g + 0.436*b
    v = 0.615*r + -0.51499*g + 0.10001*b
    return torch.cat((y,u,v), 1)

Method 4 seems to be fastest (atleast on cpu and for the size I experimented with)

**Comment by ananyamukh6** [comment]
https://github.com/kornia/kornia/pull/330#pullrequestreview-314631371
